### PR TITLE
Merge branch 'swb' into users/leish/wbmerge

### PR DESCRIPTION
**Pull request by leirocks**

### CODE REVIEW

**Comment by leirocks** [comment]
@dotnet-bot test Linux tests please

**Comment by leirocks** [comment]
@dotnet-bot test Linux tests please

**Comment by jianchun** [comment]
Commit messages could use some cleanup... Temporary private branch name does not need to appear in history. PR title should indicate this is merging master into swb.

File: lib/Runtime/Base/FunctionInfo.h [hunk 1]
```diff
@@ -44,6 +44,7 @@ namespace Js
         };
         FunctionInfo(JavascriptMethod entryPoint, Attributes attributes = None, LocalFunctionId functionId = Js::Constants::NoFunctionId, FunctionProxy* functionBodyImpl = nullptr);
         FunctionInfo(JavascriptMethod entryPoint, _no_write_barrier_tag, Attributes attributes = None, LocalFunctionId functionId = Js::Constants::NoFunctionId, FunctionProxy* functionBodyImpl = nullptr);
+        FunctionInfo(FunctionInfo& that); // Todo: (leish)(swb) find a way to prevent non-static initializer calling this ctor
```
Comment (jianchun) [inline lib/Runtime/Base/FunctionInfo.h]: How is this constructor used? Does it help to add a dummy ", _no_write_barrier_tag" parameter so it must be called explicitly as "FORCE_NO_WRITE_BARRIER_TAG(that)"?

**Pull request merged**